transport.validator.mapcss

```diff
@@ -86,39 +87,70 @@ relation[type=route_master][route_master=~/^(bus|coach|train|subway|monorail|tro
 	set pt_route_master;
 }
 
-relation.pt_route[!public_transport:version]
+way <[role="route"] relation.pt_route,
+way <[role="forward"] relation.pt_route,
+way <[role="backward"] relation.pt_route,
+way <[role="reverse"] relation.pt_route,
+way <[role="forward:stop"] relation.pt_route,
+way <[role="backward:stop"] relation.pt_route
+{
+	set pt_route_probably_v1
+}
+
+relation[type=route_master] > relation[type=route]
+{
+	set route_in_master
+}
+
+relation.route_in_master.pt_route,
+node[public_transport=stop_position] < relation.pt_route,
+*[public_transport=platform] < relation.pt_route
+{
+	set pt_route_probably_v2
+}
+
+relation.pt_route[!public_transport:version]!.pt_route_probably_v1!.pt_route_probably_v2
 {
 	throwError: tr("Missing public_transport:version tag on a public_transport route relation");
-	-osmoseItemClassLevel: "2140/21401/3";
+	-osmoseItemClassLevel: "2140/21401:0/3";
 	assertNoMatch: "relation type=route route=bus public_transport:version=1";
+	assertNoMatch: "relation type=route route=bus public_transport:version=2";
 	assertMatch: "relation type=route route=bus";
 }
 
-relation.pt_route[!network],
+relation.pt_route_probably_v1[!public_transport:version]!.pt_route_probably_v2
+{
+	throwError: tr("Missing public_transport:version tag on a Public Transport Version 1 schema relation.");
+	-osmoseItemClassLevel: "2140/21401:1/3";
+	fixAdd: "public_transport:version=1";
+}
+
+relation.pt_route_probably_v2[!public_transport:version]!.pt_route_probably_v1
+{
+	throwError: tr("Missing public_transport:version tag on a Public Transport Version 2 schema relation.");
+	-osmoseItemClassLevel: "2140/21401:2/2";
+	fixAdd: "public_transport:version=2";
+}
+
+relation.pt_route[!network][!parent_tag(network)],
```

Comment:
:confused: I'm not really up for it:
your rule is indeed more in accordance with the ptv2, but in practice, I don't think it's a good idea to leave these attributes out just because they're already in the route_master relation. In fact, these attributes are really essential to know what the transport route really refers to, and it complicates the understanding of the data to have to look for this information elsewhere (both in the actual data and in tools such as Overpass or others). What's more, route_master relations have no geometry and are therefore more difficult to query than route relations, so it seems to me to be good practice for them to carry all the information needed to identify them.
So when I'm in JOSM, this is information I always want to add to a route relation if it's missing, so I'd prefer to keep this rule as it is.

Reply:
At the most, for route relation without these tags but in a route_master relation that actually hase these tags, we could
- not have them flagged in Osmose
- have them flagged in JOSM as information (throwOther ?) istead of error

but anyway I want to keep a rule with `relation.pt_route[!network]` for JOSM

Reply:
It's your repo so I'll drop this if you want, but I can't really get on board with your mindset. Doesn't each OSM object contain a list of pointers to the relations that it's a member of? People or software who want to know information about a route and don't find it in the route relation are *supposed* to look for a possible route master relation.